File: core/src/main/java/com/pluxity/station/StationCode.java
package com.pluxity.station;

import com.pluxity.global.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.NotFound;
import org.hibernate.annotations.NotFoundAction;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class StationCode extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "station_id")
    @NotFound(action = NotFoundAction.IGNORE)
    private Station station;

    private String code;

    @Builder
    public StationCode(Station station, String code) {
        this.station = station;
        this.code = code;
    }
}

File: core/src/main/java/com/pluxity/station/StationCode.kt
package com.pluxity.station

import com.pluxity.global.entity.BaseEntity
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import org.hibernate.annotations.NotFound
import org.hibernate.annotations.NotFoundAction

@Entity
class StationCode(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
    @NotFound(action = NotFoundAction.IGNORE)
    @JoinColumn(
        name = "station_id",
    )
    @ManyToOne(fetch = FetchType.LAZY)
    var station: Station,
    var code: String,
) : BaseEntity()
